docs/API/Schedules.md

```diff
@@ -0,0 +1,146 @@
+
+```java linenums="1" title="Example Schedule"
+public class MyCustomSchedule extends Schedule/* (1)! */ {
+
+    private final int ticks, rebootSleep;
+
+    // (2)!
+    public MyCustomSchedule(ScheduleID scheduleID, ConfigurationSection instruction) throws InstructionParseException { 
+        super(scheduleID, instruction);
+        try {
+            ticks = Integer.parseInt(getTime()/* (3)! */);
+        } catch (NumberFormatException e) {
+            throw new InstructionParseException("Time is not a number");// (4)!
+        }
+
+        if (getCatchup() != CatchupStrategy.NONE) {// (5)!
+            throw new InstructionParseException("Catchup " + getCatchup() + " is not supported by this schedule type");
+        }
+
+        rebootSleep = instruction.getInt("options.rebootSleep");// (6)!
+    }
+
+    public int getTicks() { return ticks; }
+    public int getRebootSleep() { return rebootSleep; }
+}
+```
+
+1.  You can extend either `Schedule` or `CronSchedule`.   
+    The latter has already implemented all cron parsing logic.
+
+2.  You need to define a Constructor that matches **exactly** this one.  
+    **Otherwise BetonQuest can't parse your schedule!**
+
+3.  `getTime()`/`super.time` provides the raw time string.  
+    You'll need to parse it and add your own logic.  
+    _In this example we just use it as interval of ticks._
+
+4. Make sure to handle all parsing errors & thrown exceptions!  
+   If a schedule can't be loaded due to an invalid instruction,
+   you should throw a `InstructionParseException` that describes
+   the error.
+
+5.  You'll have to implement your own handling of catchup strategies
+    in the [`Scheduler`](#scheduler-class) class.  
+    **If you don't want that, you can add this check if a CatchupStrategy
+    was defined and throw an Exception.**
+
+6.  This is how you add custom options, if needed.
+
+### Scheduler Class
+
+The scheduler will receive parsed schedules using `addSchedule(S)` and hold them in the `schedules` map.  
+It should contain all the scheduling & schedule execution logic.  
+It is also responsible for catching up missed schedules, if they have a catchup strategy other than `NONE` defined.
+
+Here is a pretty basic example, that does not provide any catchup logic:
+
+```java linenums="1" title="Example Scheduler"
+public class MyCustomScheduler extends Scheduler<MyCustomSchedule>/* (1)! */ {
+
+    private List<BukkitTask> tasks;
+
+    @Override
+    public void start() {
+        super.start();// (2)!
+
+        tasks = new ArrayList<>();
+        for (MyCustomSchedule schedule : schedules.values()/* (3)! */) {
+            // (4)!
+            BukkitTask task = new BukkitRunnable() {
+                @Override
+                public void run() {
+                    executeEvents(schedule);
+                }
+            }.runTaskTimer(BetonQuest.getInstance()/* (5)! */, schedule.getRebootSleep(), schedule.getTicks());
```

Comment:
We should not use the BetonQuest instance here, This could cause problems with profilers and with exceptions, that occurs in the console.